engine/skiplist.hpp

```diff
@@ -141,7 +134,7 @@ struct SkiplistNode {
   void *heap_space_start() { return (char *)this - height * 8; }
 };
 
-class Skiplist : public PersistentList {
+class Skiplist : public PersistentList, public CompStrategy<Skiplist> {
```

Comment:
Do we really need inherit for this? I think only skiplist need a comparator

Reply:
maybe needed in the future.  This design can be independent on the skiplist, more clear.

Reply:
I think use comparator as a member is better

Reply:
the comparator is a skiplist'member now